src/main/java/com/example/bookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.user.UserRequestDto;
+import com.example.bookstore.dto.user.UserResponseDto;
+import com.example.bookstore.exception.RegistrationException;
+import com.example.bookstore.mapper.UserMapper;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.user.UserRepository;
+import com.example.bookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserMapper userMapper;
+    private final UserRepository userRepository;
+
+    @Override
+    public UserResponseDto register(UserRequestDto requestDto) throws RegistrationException {
+        if (userRepository.existsByEmail(requestDto.email())) {
+            throw RegistrationException.emailExist(requestDto.email());
```

Comment:
don't create a static method in the exception class, you won't need it anymore. it's not a class property and looks weird